core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -107,8 +111,15 @@ private MetadataColumns() {}
           ROW_POSITION.fieldId(),
           IS_DELETED.fieldId(),
           SPEC_ID.fieldId(),
-          PARTITION_COLUMN_ID);
+          PARTITION_COLUMN_ID,
+          POSITION_DELETE_TABLE_PARTITION_FIELD_ID,
```

Comment:
Why do we need a Position_Delete version of Spec_ID and File_Path, shouldn't we be able to use the original metadata columns for these?

Reply:
Yea its true, we could do that.  I think I was going back and forth whether we wanted to have that or not, as these are 'main' columns with a proper name, versus hidden columns (whose name start with _), in other words they are not the exact same column.  Im open

Reply:
Do we even have to make them metadata columns then? I thought they would be just regular columns in a table. I don't think they should be added to `META_COLUMNS`. I think metadata columns should be only about columns we can project on demand. That's why we did not add changelog columns here.

Let me also think about reserving field IDs for them. It is similar yet different use case compared to changelog columns as there is no changelog table as such.

Reply:
Removed from list of metadata columns. It was from an earlier change where I was trying to re-use existing Spark RowReader and it checked that projected column must either be part of schema, or metadata column.

Kept the field ids in this file to reserve them, to avoid conflict with "row" struct of this table, which has the data table schema.